### ISSUE TITLE
Proxy Wasm Go SDK is deprecated

### ISSUE THREAD

**Issue by acouvreur** [sablierapp/sablier#426]
ProxyWasm go SDK was archived: https://github.com/tetratelabs/proxy-wasm-go-sdk

> [!WARNING] 
> We are no longer recommending this SDK or Wasm in general for anyone due to the fundamental memory issue of TinyGo (See [the detailed explanation](https://github.com/tetratelabs/proxy-wasm-go-sdk/issues/450#issuecomment-2253729297) by a long-time community member)
> as well as [the project state of Proxy-Wasm in general](https://github.com/envoyproxy/envoy/issues/35420).
> If you are not in a position where you have to run untrusted binaries (like for example, you run Envoy proxies while your client gives you the binaries to run), we recommend using other extension mechanism 
> such as Lua or External Processing which should be comparable or better or worse depending on the use case. 
> 
> If you are already using this SDK, but still want to continue using Wasm for some reason instead of Lua or External Processing, 
> we strongly recommend migrating to the Rust or C++ SDK due to the memory issue of TinyGo described in the link above.
> 
> We keep this repository open and not archived for the existing users, but we cannot provide any support or guarantee for the future development of this SDK.
> However, at any time, we may decide to archive this repository if we see no reason to keep it open.

**Comment by davidspek** [comment]
Still active fork: https://github.com/higress-group/proxy-wasm-go-sdk

**Comment by acouvreur** [comment]
> Still active fork: https://github.com/higress-group/proxy-wasm-go-sdk

I think that because of the argument about why it is no longer recommended I'd prefer to rewrite it in Rust.

**Comment by thiagogcm** [comment]
there's a new "official" fork https://github.com/proxy-wasm/proxy-wasm-go-sdk

**Comment by acouvreur** [comment]
That's interesting, I will try porting the actual plugin to the new fork as they seem to need active users for this.

**Comment by martijneken** [comment]
Yeah! We're excited about it. Note the key difference: this is full go, not tiny go. That solves a ton of problems with memory management and capabilities. It depends on not-yet-mainstream Go 1.24. Please let us know how it works for you! CC @leonm1

**Comment by acouvreur** [comment]
Just tried to use your new proxy-wasm implementation with the release of go 1.24 in #521 and got the following issue:

```
 Error:     | [2025-02-15 01:48:18.173][1][error][wasm] [source/extensions/common/wasm/wasm_vm.cc:38] Failed to load Wasm module due to a missing import: wasi_snapshot_preview1.sched_yield
Error:     | [2025-02-15 01:48:18.173][1][error][wasm] [source/extensions/common/wasm/wasm.cc:110] Wasm VM failed Failed to initialize Wasm code
```

See here for more details:
https://github.com/sablierapp/sablier/actions/runs/13340566899/job/37264387799?pr=521#step:8:617

I swapped it as a drop in replacement, tell me if that's not supposed to be the case.

**Comment by martijneken** [comment]
The missing hostcalls were added to CppHost (as noops) in https://github.com/proxy-wasm/proxy-wasm-cpp-host/pull/427 and Envoy in https://github.com/envoyproxy/envoy/pull/37868. They don't need to do anything but they do need to exist, so I expect we just need to update ProxyWasmCppHost in nginx.

**Comment by acouvreur** [comment]
Completed in #521 

The ProxyWasm plugin now uses https://github.com/proxy-wasm/proxy-wasm-go-sdk
